src/dune_rules/link_time_code_gen.ml

```diff
@@ -7,32 +7,25 @@ type t =
 
 let generate_and_compile_module cctx ~precompiled_cmi ~name ~lib ~code ~requires
     =
-  let open Resolve.Memo.O in
-  let* module_ =
-    let gen_module =
-      let src_dir =
-        let obj_dir = Compilation_context.obj_dir cctx in
-        Obj_dir.obj_dir obj_dir
-      in
-      Module.generated ~kind:Impl ~src_dir name
+  let sctx = Compilation_context.super_context cctx in
+  let open Memo.O in
+  let* modules = Dir_contents.modules_of_lib sctx lib in
+  let modules =
```

Comment:
probably a good idea to wait for https://github.com/ocaml/dune/pull/6605, then we can remove this error message